src/lfortran/semantics/ast_common_visitor.h

```
@@ -4294,6 +4297,16 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                 }
             }
             if( a.m_right ) {
+                if (ASRUtils::is_array(ASRUtils::expr_type(a.m_right))) {
+                    is_arg_array = true;
+                    if (ASRUtils::is_array(ASRUtils::expr_type(a.m_right))) {
+                        ASR::dimension_t* arg_dim = nullptr;
+                        LCOMPILERS_ASSERT(
+                            ASRUtils::extract_dimensions_from_ttype(
+                                ASRUtils::expr_type(a.m_right), arg_dim) == 1);
+                        res_dims_vec.push_back(al, arg_dim[0]);
```

Let's throw a beautiful error message here instead of the assertion failure when the user passes a multi-dimensional index array.